### ISSUE TITLE
Template Processor does not account for any variables defined in a nested template in a deployments resource

### ISSUE THREAD

**Issue by yane3628** [Azure/template-analyzer#39]
``` javascript
{
  "name": "string",
  "type": "Microsoft.Resources/deployments",
  "apiVersion": "2020-10-01",
  "location": "string",
  "scope": "string",
  "subscriptionId": "string",
  "resourceGroup": "string",
  "properties": {
    "expressionEvaluationOptions": {
      "scope": "inner"
    },
    "template": {<WHERE THIS NESTED TEMPLATE HAS ITS OWN VARIABLES DEFINED>},
}
```

Error thrown is: "The template variable '<variable_name>' is not found."
This depends on a setting to set the scope of the inner template: https://docs.microsoft.com/en-us/azure/templates/microsoft.resources/deployments?tabs=json#expressionevaluationoptions-object

**Comment by norbusonam** [comment]
@yane3628 @j0tr I am not sure how to implement this, but bicep produces these nested templates. Therefore, in my bicep checks, problems are going unnoticed.

**Comment by anicetdushime** [comment]
The nested template support does support accounting for variables, parameters and functions defined inside a nested template, which should handle this issue

**Comment by VeraBE** [comment]
@anicetdushime Let's keep this open until your PR is merged. If you write "Closes #39" in the description of your PR then both will get linked
